resolution/design/00_resolution.md

```diff
@@ -0,0 +1,331 @@
+This enables resolution convoluted fitting to be performed by replacing, e.g., the call to `multfit_sqw()` in the [existing `multifit` solution] with a call to a different gateway function `tobyfit()`:
+
+```matlab
+obj = tobyfit(sqw_obj1 {, sqw_obj2, ...}); % object containing 1+ sqw object(s)
+obj = obj.set_fun(@user_sqw_function, p_i); % define a foreground function
+obj = obj.set_bfun(@user_bkg_function, p_j); % define a background function
+obj = obj.set_free(...); % optionally fix foreground parameters
+obj = obj.set_bfree(...); % optionally fix background parameters
+obj = obj.set_bind(...); % optionally bind foreground parameters
+obj = obj.set_bbind(...); % optionally bind background parameters
+[sim_obj(s), p_fit] = obj.fit();  
+```
+
+The flexibility provided by the interaction of `mfclass`, `mfclass_tobyfit`, `mfclass_wrapfun`, and `mfclass_plist` comes with the significant drawback that the code is difficult to understand and therefore difficult to maintain.
+
+
+[`OptModel`]: ../../optimisation/design/Model_Optimisation_Design.md#optimisation-model-object
+[existing `multifit` solution]: ../../optimisation/design/Model_Optimisation_Design.md#existing-solution
+[mfclass]: images/mfclass_diagram.png
+
+# Proposed Solution
+
+As part of the optimisation scheme a new class [`OptFunction`] will be created; objects of this class are not intended to handle instrumental resolution convolution.
+Instead resolution effects will be dealt with by extending `OptFunction` with one or more new classes.
+
+There has been some discussion of implementing new methods for estimating the instrumental resolution of a direct-geometry time-of-flight spectrometer, e.g., making use of Monte Carlo ray-tracing to produce a probability distribution of neutron states incident on the sample along with analytic approximations to the secondary spectrometer.
+
+Additionally, we have investigated new methods for performing the resolution convolution -- notably, reuse of in-resolution function evaluations for nearby detector pixels.
+
+In order to support multiple ways of estimating the instrument resolution and performing the convolution we require a flexible resolution convolution scheme to be implemented in conjunction with `OptFunction`.
+
+## `OptFunction` subclasses
+`OptFunction` will be extended to include an `initialise()` method which performs no action by default, and a `simulate()` method which calls the `initialise()` method before the `evaluate()` method. Same-named similar extensions will be added to `OptFunctions` as well.
+The two 'gateway' methods, `simulate()` and `optimise()`, in `OptModel` will be modified to first call its contained `OptFunctions` objects' `initialise()` and then to continue utilising a private `evaluate()` method to finish simulating or optimising the model.
+The `evaluate()` method of `OptFunction` and `OptFunctions` must remain public.
```

Comment:
I was not aware of class access lists in MATLAB, but I am very happy to learn of the feature.

`OptFunction` and `OptFunctions` need to have a `simulate()` method in case `initialise()` is not an empty function *and* a user wants to evaluate their (possibly resolution convoluted) function or functions-with-a-reduction-operator *without* worrying about setting up everything for optimising the function/functions to measured data.

Removing `initialise()` and `evaluate()` from their public methods is certainly a good idea.
I'll modify the document to reflect this.